sources/packages/web/src/components/common/OverawardDetails.vue

```diff
@@ -0,0 +1,225 @@
+                </span>
+              </template></Column
+            >
+          </DataTable>
+        </toggle-content>
+      </content-group>
+    </v-container>
+  </v-card>
+  <add-manual-overaward-deduction
+    ref="addManualOverawardDeduction"
+    :allowedRole="Role.StudentAddOverawardManual"
+  />
+</template>
+<script lang="ts">
+import { ref, onMounted, defineComponent, computed } from "vue";
+import { OverawardService } from "@/services/OverawardService";
+import {
+  DEFAULT_PAGE_LIMIT,
+  PAGINATION_LIST,
+  DEFAULT_PAGE_NUMBER,
+  DisbursementOverawardOriginType,
+  Role,
+} from "@/types";
+import { useFormatters, ModalDialog, useSnackBar } from "@/composables";
+import {
+  OverawardAPIOutDTO,
+  OverawardManualRecordAPIInDTO,
+} from "@/services/http/dto";
+import CheckPermissionRole from "@/components/generic/CheckPermissionRole.vue";
+import AddManualOverawardDeduction from "@/components/aest/students/modals/AddManualOverawardDeduction.vue";
+
+export default defineComponent({
+  components: { CheckPermissionRole, AddManualOverawardDeduction },
+  emits: {
+    updateOverawardDetails: null,
+  },
+  props: {
+    studentId: {
+      type: Number,
+      required: true,
+    },
+    showAddedBy: {
+      type: Boolean,
+      required: false,
+      default: false,
+    },
+    allowManualOverawardDeduction: {
+      type: Boolean,
+      required: false,
+      default: false,
+    },
+  },
+  setup(props, context) {
+    const page = ref(DEFAULT_PAGE_NUMBER);
+    const pageLimit = ref(DEFAULT_PAGE_LIMIT);
+    const { dateOnlyLongString, formatCurrency } = useFormatters();
+    const snackBar = useSnackBar();
+    const overawardDetails = ref([] as OverawardAPIOutDTO[]);
+    const addManualOverawardDeduction = ref(
+      {} as ModalDialog<OverawardManualRecordAPIInDTO | boolean>,
+    );
+    const overawards = computed(() =>
+      overawardDetails.value.filter(
+        (overaward) =>
+          ![
+            DisbursementOverawardOriginType.AwardDeducted,
+            DisbursementOverawardOriginType.ManualRecord,
+          ].includes(overaward.overawardOrigin),
+      ),
+    );
+
+    const overawardDeductions = computed(() =>
+      overawardDetails.value.filter((overaward) =>
+        [
+          DisbursementOverawardOriginType.AwardDeducted,
+          DisbursementOverawardOriginType.ManualRecord,
+        ].includes(overaward.overawardOrigin),
+      ),
+    );
+
+    const addManualOveraward = async () => {
+      const manualOveraward =
+        await addManualOverawardDeduction.value.showModal();
+      if (manualOveraward) {
+        try {
+          await OverawardService.shared.addManualOverawardDeduction(
+            props.studentId,
+            manualOveraward as OverawardManualRecordAPIInDTO,
+          );
+          snackBar.success("Overaward deduction added successfully.");
+          context.emit("updateOverawardDetails");
```

Comment:
I believe that the event must be named after what it represents, for instance, `overawardBalanceUpdated` or `manualOverawardAdded`. `updateOverawardDetails` is meaningful for the consumer of the event only.